### PR TITLE
Install xz's dll in bin

### PR DESCRIPTION
**Pull request by danlipsa**

### CODE REVIEW

**Comment by danlipsa** [comment]
@johnwparent Please review

**Comment by danlipsa** [comment]
@johnwparent any idea what is wrong with ci/gitlab-ci? Thanks!

**Comment by johnwparent** [comment]
@spackbot run pipeline

File: var/spack/repos/builtin/packages/xz/package.py [hunk 1]
```diff
@@ -112,17 +112,31 @@ def msbuild_args(self):
     def install(self, pkg, spec, prefix):
         with working_dir(self.build_directory):
             # Ensure we have libs directory
-            mkdirp(prefix.lib)
+            if not os.path.isdir(prefix.lib):
+                mkdirp(prefix.lib)
+            if not os.path.isdir(prefix.bin):
+                mkdirp(prefix.bin)
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/xz/package.py]: `mkdirp` works whether or not dir has already been made: please remove from `if` statement.

File: var/spack/repos/builtin/packages/xz/package.py [hunk 2]
```diff
@@ -112,17 +112,31 @@ def msbuild_args(self):
...
             libs_to_find = []
-            if "libs=shared" in self.pkg.spec:
```
Comment (scheibelp) [inline var/spack/repos/builtin/packages/xz/package.py:117]: Less-redundant rewrite:

```
if self.pkg.spec.satisfies("libs=shared"):
  dlls_to_find.append(...)
if self.pkg.spec.satisfies("libs=static"): #this is not an elif
  libs_to_find.append(...)
```

**Review by scheibelp — CHANGES_REQUESTED** [review]
I have a couple minor requests

**Review by scheibelp — CHANGES_REQUESTED** [review]
I have a couple minor requests

**Comment by danlipsa** [comment]
@scheibelp Thanks for the review. All changes you requested have been pushed.

**Comment by danlipsa** [comment]
@johnwparent Any ideas what is wrong with the pipeline that failed? Thanks!

**Comment by scheibelp** [comment]
spackbot run pipeline

**Comment by scheibelp** [comment]
@spackbot run pipeline

**Comment by kwryankrattiger** [comment]
This is stuck on a generate issue that is thinking it found or could prune xz from rebuilding. Maybe this needs a rebase on the latest?

@scottwittenburg fyi

**Comment by scottwittenburg** [comment]
It seems that `xz/rv46efa` is truly not present on any configured mirror.  But when I attempt to reproduce this pipeline generation generation issue locally, I get different behavior.  Specifically, `xz/rv46efa` along with many other specs are scheduled for rebuild.  This is just what happened in another attempt to reproduce a similar issue a couple weeks ago.  In that case, I asked spackbot to re-run the pipeline, and suddenly all the missing jobs were generated.

Let's see if that happens again here.

**Comment by scottwittenburg** [comment]
@spackbot run pipeline

**Comment by scottwittenburg** [comment]
Ugh, so this time `xz/rv46efa` and other dependents were scheduled to be built.  This makes me think that #38185 isn't going to help the way it is, since asking to run again with extra environment variables may not result in the same behavior.  Instead, as we discussed on the call today, I'll just print more information in the staging summary by default.

**Comment by danlipsa** [comment]
@scheibelp Please review. All tests pass now.

**Comment by danlipsa** [comment]
@scheibelp This passes all tests. Please review.

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**